### ISSUE TITLE
Deep copy object with two dates that are the same doesn't work

### ISSUE THREAD

**Issue by pjm17971** [sasaplus1/deepcopy.js#7]
I noticed this was causing a problem in some of our code. We've worked around it, but here's the example:

<img width="872" alt="screen shot 2015-12-10 at 1 58 31 pm" src="https://cloud.githubusercontent.com/assets/1288813/11730604/0ca3660e-9f4b-11e5-87a0-e39683bfec21.png">

On the left is the original data. On the right is the deepcopy. Notice that birthday.yyy is now an empty object. This seems to be because they birthday.xxx and birthday.yyy are references to the same Date. If they are different Dates, it works.

**Comment by sasaplus1** [comment]
@pjm17971

Thanks your bug report. I will fix it later.
If you are hurry, use below:

```diff
diff --git a/deepcopy.js b/deepcopy.js
index f583ad0..13ce1f8 100644
--- a/deepcopy.js
+++ b/deepcopy.js
@@ -197,6 +197,7 @@
         if (idx === -1) {
           // not circular reference
           obj = (util.isArray(val)) ? [] : {};
+          obj = (util.isDate(val)) ? val : {};
 
           visited.push(val);
           reference.push(obj);
```

**Comment by sasaplus1** [comment]
fixed version 0.6.0.
